src/qat/purr/backends/echo.py

```diff
@@ -76,8 +76,8 @@ def apply_setup_to_hardware(
                         frequency=5.5e9,
                         scale=0.0,
                     )
-                qubit.add_coupled_qubit(qubit_devices[(i + 1) % qubit_count])
-                qubit.add_coupled_qubit(qubit_devices[(i - 1) % qubit_count])
+            qubit.add_coupled_qubit(qubit_devices[(i + 1) % qubit_count])
+            qubit.add_coupled_qubit(qubit_devices[(i - 1) % qubit_count])
```

Comment:
can't believe it, was it adding to a set or a list ?

Reply:
The `Qubit` has a method that actually does nothing if it already has that coupling, but we were still going into that method and checking it an equivalent coupling already exists way more times than we needed. :loop: